### PR TITLE
Invalid process checker

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: invalidprocess/main.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	store := rocksdb.New(globaldb.GetRocksDB())
+	defer globaldb.GetRocksDB().Close()
+
+	var idsToDelete []string
+	err := store.Walk(ctx, func(obj *storage.ProcessIndicator) error {
+		var bad bool
+		if !utf8.ValidString(obj.GetSignal().GetExecFilePath()) {
+			bad = true
+			log.Errorf("Found non utf-8 in exec file path for deployment=%s container=%s data=%s", obj.GetDeploymentId(), obj.GetContainerName(), obj.GetSignal().GetExecFilePath())
```
Comment (rhybrillou) [inline invalidprocess/main.go:32]: Would it make sense to extract these error logs into a common function ?
```
func logBadProcessIndicator(pi *storage.ProcessIndicator, badField string, badValue string) {
    log.Errorf("Found non utf-8 in %s for deployment=%s container=%s data=%q", badField, pi.GetDeploymentId(), pi.GetContainerName(), badValue)
}
```

File: invalidprocess/main.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+)
+
+var log = logging.LoggerForModule()
+
+func main() {
+	shouldDelete := os.Getenv("DELETE_PROCESSES") == "true"
+
+	ctx := sac.WithAllAccess(context.Background())
+
+	store := rocksdb.New(globaldb.GetRocksDB())
```
Comment (rhybrillou) [inline invalidprocess/main.go:24]: Is it a pre-requisite for some postgres migrations to clean the processInfo data in rocksDB ?
Is there a chance that process indicators in postgres could be impacted too ?

Comment (connorgorman) [reply invalidprocess/main.go:24]: This is not going to be shipped

File: invalidprocess/main.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	})
+	if err != nil {
+		log.Fatalf("error walking processes: %v", err)
+	}
+	log.Infof("found %d processes that are not properly UTF8 encoded", len(idsToDelete))
+	if !shouldDelete {
+		return
+	}
+
+	if err := store.DeleteMany(ctx, idsToDelete); err != nil {
```
Comment (rhybrillou) [inline invalidprocess/main.go:61]: What is the expected number of process indicators to delete ?
Would it make sense to split the initial list in smaller chunks ?

**Review by rhybrillou — APPROVED** [review]

**Pull request closed without merging**